src/app/library-detail/library-detail.component.ts

```diff
@@ -42,9 +42,11 @@ export class LibraryDetailComponent implements OnInit {
         this.pagination = {currentPage: 0, itemsPerPage: 30, totalItems: 0, totalPages: 1};
       }
       this.pagination.currentPage = parseInt(page, 10);
+    } else {
+      this.pagination = {currentPage: 0, itemsPerPage: 30, totalItems: 0, totalPages: 1};
```

Comment:
We can remove this as a recent update made by me will make it so passing no pagination will default to 30 pages.